src/include/data/data_repository.hpp

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2025, Sirius Contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+#include "data_batch.hpp"
+#include "helper/helper.hpp"
+
+namespace sirius {
+
+class IDataRepository {
+public:
+    virtual void AddNewDataBatch(sirius::unique_ptr<DataBatchView> data_batch) = 0;
+
+    virtual sirius::unique_ptr<DataBatchView> EvictDataBatch() = 0;
```

Comment:
I am not sure we have defined how eviction happens in our current execution design. So for example it is possible that we know upfront every view that will be made from a databatch, we create them all, when refcount = 0 then we evict, or it might be the case that we don't know how many tasks will need access to that batch. If the later is the case then we need some way of knowing which batch to evict. and eviction should probably not return a view. If we are using status it should return a status, if we are using exceptions it should return void or throw

Reply:
also now that i think about it there is no such thing as eviction from the DataRepository right? The data repository has a view that will only be used once unless I am mistaken, the Data Batch will resides in the Data Repository Manager Class

Reply:
What i mean by eviction is when a task pulling the data batch out of that repository

Reply:
If I understand correctly, I think you are talking about evicting from GPU memory